libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -467,15 +467,7 @@ - (void)replayRequest:(SFRestRequest *)request response:(NSURLResponse *)respons
             } error:^(NSError *refreshError) {
                 __strong typeof(weakSelf) strongSelf = weakSelf;
                 [SFSDKCoreLogger e:[strongSelf class] format:@"Failed to refresh expired session. Error: %@", refreshError];
-                
-                // Call the failure block for the triggering request first
-                if (request.failureBlock) {
```

Comment:
For line 478, I saw an issue where the same request is being removed multiple times along with the callback being executed more than once (if the active queue is at least two requests), I'm thinking it's because `flushPendingRequestQueue` is synchronized to self but the `removeObject` call isn't.

The part I don't know the history on is why the failure block for first request was triggered separately, it looks like the same values are passed to the failure block whether it's called here or through `flushPendingRequestQueue`